### ISSUE TITLE
[java] AddEmptyString false positive in annotation value

### ISSUE THREAD

**Issue by davidburstrom** [pmd/pmd#2364]
**Affects PMD Version:** 

6.22 and earlier

**Rule:**

AddEmptyString

**Description:**

When using annotations that require string values, e.g. `SuppressWarnings`, and the value should be computed from e.g. an integer constant, empty strings addition is the (only?) way to get a compile time constant string. It is desirable to do so in order to adhere to the DRY principle.
That is, the rule needs to consider its context to determine correctness.

**Code Sample demonstrating the issue:**

```
static final int MY_CONST = 23;
@SomeAnnotationWithStringValue("" + MY_CONST + "")
Object object;
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

N/A
